### DISCUSSION TITLE
npm run build 的时候报错有 dead links，应该怎么修复呢？

### DISCUSSION THREAD

**Discussion by 674019130** [YunYouJun/valaxy#390] [Q&A] [answered]
╭────────────────────────╮
 │                        │
 │  🌠 Start building...  │
 │                        │
 ╰────────────────────────╯

i use vite-ssg to do ssg build...

[vite-ssg] Build for client...

The language 'mysql' is not loaded, falling back to 'txt' for syntax highlighting.

The language 'mysql' is not loaded, falling back to 'txt' for syntax highlighting.

The language 'mysql' is not loaded, falling back to 'txt' for syntax highlighting.

The language 'mysql' is not loaded, falling back to 'txt' for syntax highlighting.

The language 'mysql' is not loaded, falling back to 'txt' for syntax highlighting.

The language 'mysql' is not loaded, falling back to 'txt' for syntax highlighting.

The language 'mysql' is not loaded, falling back to 'txt' for syntax highlighting.

 ERROR  Dead links found in E:/BlogCode/valaxyBlogCode/valaxy-blog/pages/posts/Designing Data-Intensive Applications 读书笔记 0x04.md

 ERROR  [ { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x04.md' },
  { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x04.md' },
  { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x04.md' },
  { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x04.md' },
  { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x04.md' },
  { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x04.md' } ]

 ERROR  Dead links found in E:/BlogCode/valaxyBlogCode/valaxy-blog/pages/posts/Designing Data-Intensive Applications 读书笔记 0x03.md

 ERROR  [ { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x03.md' },
  { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x03.md' },
  { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x03.md' } ]

 ERROR  Dead links found in E:/BlogCode/valaxyBlogCode/valaxy-blog/pages/posts/Designing Data-Intensive Applications 读书笔记 0x02.md

 ERROR  [ { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x02.md' },
  { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x02.md' } ]

 ERROR  Dead links found in E:/BlogCode/valaxyBlogCode/valaxy-blog/pages/posts/Designing Data-Intensive Applications 读书笔记 0x01.md

 ERROR  [ { url: './index', file: 'posts/Designing Data-Intensive Applications 读书笔记 0x01.md' } ]

node_modules/.pnpm/valaxy-theme-yun@0.18.6/node_modules/valaxy-theme-yun/components/third/YunArtalk.vue?vue&type=script&setup=true&lang.ts (14:14): "useArtalkWithOptions" is not exported by "node_modules/.pnpm/valaxy@0.18.6_@babel+parser@7.24.5_@types+markdown-it@14.0.1_@types+node@20.12.7_@unhead+shar_qhm4uwb2fsrmt7m7yb7jyneeri/node_modules/valaxy/client/addons/index.ts", imported by "node_modules/.pnpm/valaxy-theme-yun@0.18.6/node_modules/valaxy-theme-yun/components/third/YunArtalk.vue?vue&type=script&setup=true&lang.ts".
node_modules/.pnpm/valaxy-theme-yun@0.18.6/node_modules/valaxy-theme-yun/components/third/YunTwikoo.vue?vue&type=script&setup=true&lang.ts (22:14): "useTwikooWithOptions" is not exported by "node_modules/.pnpm/valaxy@0.18.6_@babel+parser@7.24.5_@types+markdown-it@14.0.1_@types+node@20.12.7_@unhead+shar_qhm4uwb2fsrmt7m7yb7jyneeri/node_modules/valaxy/client/addons/index.ts", imported by "node_modules/.pnpm/valaxy-theme-yun@0.18.6/node_modules/valaxy-theme-yun/components/third/YunTwikoo.vue?vue&type=script&setup=true&lang.ts".
node_modules/.pnpm/valaxy-theme-yun@0.18.6/node_modules/valaxy-theme-yun/components/third/YunAlgoliaSearch.vue?vue&type=script&setup=true&lang.ts (21:53): "useAddonAlgolia" is not exported by "node_modules/.pnpm/valaxy@0.18.6_@babel+parser@7.24.5_@types+markdown-it@14.0.1_@types+node@20.12.7_@unhead+shar_qhm4uwb2fsrmt7m7yb7jyneeri/node_modules/valaxy/client/addons/index.ts", imported by "node_modules/.pnpm/valaxy-theme-yun@0.18.6/node_modules/valaxy-theme-yun/components/third/YunAlgoliaSearch.vue?vue&type=script&setup=true&lang.ts".
x Build failed in 3.78s

 ERROR  [vite-ssg] An internal error occurred.

Error: [valaxy:loader] One or more pages contain dead links.
    at Object.renderStart (file:///E:/BlogCode/valaxyBlogCode/valaxy-blog/node_modules/.pnpm/valaxy@0.18.6_@babel+parser@7.24.5_@types+markdown-it@14.0.1_@types+node@20.12.7_@unhead+shar_qhm4uwb2fsrmt7m7yb7jyneeri/node_modules/valaxy/dist/chunk-VQ75V4CX.mjs:74:1000)
    at file:///E:/BlogCode/valaxyBlogCode/valaxy-blog/node_modules/.pnpm/rollup@4.17.2/node_modules/rollup/dist/es/shared/node-entry.js:19778:40
    at async Promise.all (index 0)
    at async PluginDriver.hookParallel (file:///E:/BlogCode/valaxyBlogCode/valaxy-blog/node_modules/.pnpm/rollup@4.17.2/node_modules/rollup/dist/es/shared/node-entry.js:19706:9)
    at async Bundle.generate (file:///E:/BlogCode/valaxyBlogCode/valaxy-blog/node_modules/.pnpm/rollup@4.17.2/node_modules/rollup/dist/es/shared/node-entry.js:18136:13)
    at async file:///E:/BlogCode/valaxyBlogCode/valaxy-blog/node_modules/.pnpm/rollup@4.17.2/node_modules/rollup/dist/es/shared/node-entry.js:20696:27
    at async catchUnfinishedHookActions (file:///E:/BlogCode/valaxyBlogCode/valaxy-blog/node_modules/.pnpm/rollup@4.17.2/node_modules/rollup/dist/es/shared/node-entry.js:20123:16) {
  code: 'PLUGIN_ERROR',
  plugin: 'valaxy:loader',
  hook: 'renderStart'
}
o Run build:after hooks

i RSS Generating ...
i RSS Site Url: https://674019130.github.io/
√ [rss] dist: E:\BlogCode\valaxyBlogCode\valaxy-blog\dist/feed.xml
√ [rss] public: E:\BlogCode\valaxyBlogCode\valaxy-blog\dist/feed.xml
√ [atom] dist: E:\BlogCode\valaxyBlogCode\valaxy-blog\dist/atom.xml
√ [atom] public: E:\BlogCode\valaxyBlogCode\valaxy-blog\dist/atom.xml
√ [json] dist: E:\BlogCode\valaxyBlogCode\valaxy-blog\dist/feed.json
√ [json] public: E:\BlogCode\valaxyBlogCode\valaxy-blog\dist/feed.json

**Comment by fanghsiu** [comment, marked as answer]
```ts
// valaxy.config.ts
ignoreDeadLinks: true,
```

**Reply by 674019130** [reply]
妈妈你看，GitHub 里有超人:superman

**Reply by 674019130** [reply]
实际上我的解决方案是检查了图片和其他所有引用的超链接，部分 GitBook 的图片直接复制到 Typora 确实会出现能加载但是无法访问的情况，懒得一个个用图床迁移，也暂时懒得配 Typora 的图床了，直接删掉不能访问的链接之后 build 成功了。
config 配置待验证 0.0
